### PR TITLE
[android][expo-random] Rewrite android code to Kotlin

### PR DESCRIPTION
**Pull request by kkafar**
# Why

This task was done as a part of migrating expo modules to Kotlin project.

# How

The code had been initially converted from Java to Kotlin with help of Android Studio's conversion tool. Then it was manually adjusted / improved.

* Changed visibility modifier of `mSecureRandom` property to `private`
* Replaced null-check & initialisation with lazy evaluated value

If `mSecureRandom` was intentionally left as not-`private` then above changes should not be applied

# Test Plan

BareExpo (test-suite + NCL)

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.io and README.md).
- [x] This diff will work correctly for `expo build` (eg: updated `@expo/xdl`).

### CODE REVIEW

**Review by barthap — COMMENTED** [review]
Just one suggestion, it's optional and totally up to you 😄

**Review by barthap — APPROVED** [review]

File: packages/expo-random/android/src/main/java/expo/modules/random/RandomModule.kt [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+import java.security.SecureRandom
+
+import com.facebook.react.bridge.Promise
+import com.facebook.react.bridge.ReactApplicationContext
+import com.facebook.react.bridge.ReactContextBaseJavaModule
+import com.facebook.react.bridge.ReactMethod
+
+class RandomModule(context: ReactApplicationContext) : ReactContextBaseJavaModule(context) {
+  private val secureRandom: SecureRandom by lazy { SecureRandom() }
```
Comment (lukmccall) [inline packages/expo-random/android/src/main/java/expo/modules/random/RandomModule.kt:13]:
```suggestion
  private val secureRandom by lazy { SecureRandom() }
```
The type here is unnecessary, but is up to you 😉

**Review by lukmccall — APPROVED** [review]
LGTM 🚀

File: packages/expo-random/android/src/main/java/expo/modules/random/RandomModule.kt (hunk 1 above)
Comment (kkafar) [reply packages/expo-random/android/src/main/java/expo/modules/random/RandomModule.kt:13]: Type of this property is clear anyway, so let's go for it.

**Review by bbarthec — APPROVED** [review]
Looks good 👍

**Pull request merged**